openedx/core/djangoapps/user_api/preferences/api.py

```diff
@@ -142,7 +142,9 @@ def update_user_preferences(requesting_user, update, user=None):
         if preference_value is not None:
             try:
                 serializer = serializers[preference_key]
-                serializer.save()
+
+                if serializer.instance is None or serializer.instance.value != serializer.validated_data['value']:
+                    serializer.save()
```

Comment:
I think this answers one of my questions from above.  It doesn't add queries because the actual save is now conditional on data having actually changed. Nice.